src/psyclone/tests/psyir/transformations/intrinsics/matmul2code_trans_test.py

```diff
@@ -507,64 +531,70 @@ def test_validate11():
     matrix.children[2] = my_range
     with pytest.raises(TransformationError) as excinfo:
         trans.validate(matmul)
-    assert ("To use matmul2code_trans on matmul, only the first two indices "
-            "of the 1st argument are permitted to be Ranges but "
-            "found Range at index 2." in str(excinfo.value))
+    assert ("Transformation Error: To use matmul2code_trans on matmul, "
+            "each Range index of the argument 'x' must be a full range "
+            "but found non full range at position 2." in str(excinfo.value))
 
 
-def test_validate12():
+def test_validate13():
     '''Check that the Matmul2Code validate method raises the expected
     exception when the supplied node is a MATMUL IntrinsicCall but
-    the first dimension of its second (vector) argument is not a full
-    range.
+    a dimension of the second (vector) argument is indexed via a partial range.
 
     '''
     trans = Matmul2CodeTrans()
     matmul = create_matmul()
     vector = matmul.arguments[1]
-    vector.children[0] = Literal("1", INTEGER_TYPE)
+    my_range = vector.children[0].copy()
+    vector.children[2] = my_range
     with pytest.raises(TransformationError) as excinfo:
         trans.validate(matmul)
-    assert ("To use matmul2code_trans on matmul, the first index of the 2nd "
-            "argument 'y' must be a full range." in str(excinfo.value))
+    assert ("Transformation Error: To use matmul2code_trans on matmul, "
+            "each Range index of the argument 'y' must be a full range "
+            "but found non full range at position 2." in str(excinfo.value))
 
 
-def test_validate_2nd_dim_2nd_arg():
-    ''' Check that the Matmul2Code validate method raises the expected
-    exception when the second dimension of the second argument to MATMUL
-    is not a full range. '''
+def test_validate14():
```

Comment:
You don't have to do this now but I would stay away from numbering tests like this. Rupert used to like doing it this way but I found that it was a pain when you came back and wanted to add new tests. It's also nice to have informative names.

Reply:
Addressed in 10d5abbfb758dbaa45439829f974a99504e27227. I tried to rename all the numbered validation tests into something that their docstring describes.

Reply:
Thanks for all that renaming. Above and beyond :-)